### ISSUE TITLE
redudant_clone does not lint on async function calls

### ISSUE THREAD

**Issue by irexiz** [rust-lang/rust-clippy#5827]
I tried this code:

```rust
#[tokio::main]
async fn main() -> Result<(), &'static str> {
    let x = String::from("abc");
    foo_async(&x.clone()).await;
    Ok(())
}
async fn foo_async(s: &str) {
    println!("{}", s);
}
```

 *I expected to have Clippy inform me about having a redundant clone, with this warning:*
```bash
warning: redundant clone
 --> src/main.rs:5:16
  |
5 |     foo_sync(&x.clone());
  |                ^^^^^^^^ help: remove this
  |
  = note: `#[warn(clippy::redundant_clone)]` on by default
note: cloned value is neither consumed nor mutated
 --> src/main.rs:5:15
  |
5 |     foo_sync(&x.clone());
  |               ^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_clone

warning: 1 warning emitted
```
*When using an async function call, Clippy outputs no warnings despite evidently having a redundant clone*
```bash
    Finished dev [unoptimized + debuginfo] target(s) in 14.10s
```

Full code:
```rust
#[tokio::main]
async fn main() -> Result<(), &'static str> {
    let x = String::from("abc");
    foo_async(&x.clone()).await;
    //foo_sync(&x.clone());
    Ok(())
}
async fn foo_async(s: &str) {
    println!("{}", s);
}

//fn foo_sync(s: &str) {
//println!("{}", s);
//}
```

### Meta

- `clippy 0.0.212 (346aec9 2020-07-11)`
- `rustc -Vv`:
```bash
rustc 1.46.0-nightly (346aec9b0 2020-07-11)
binary: rustc
commit-hash: 346aec9b02f3c74f3fce97fd6bda24709d220e49
commit-date: 2020-07-11
host: x86_64-unknown-linux-gnu
release: 1.46.0-nightly
LLVM version: 10.0
```

<details><summary>Backtrace</summary>
  <p> 
  </p>
</details>

**Comment by piotrmaks** [comment]
It's enough to await a future to silent the warning.
```rust
#[derive(Clone)]
struct Cloneable;

async fn bar() {}

pub async fn silent() {
    let c = Cloneable;
    let _ = c.clone();
    let _ = bar().await;
}
```

 This is happening on both stable and nightly rust.

_I expected to have Clippy inform me about having a redundant clone, with this warning_:
```rust
warning: redundant clone
 --> src/lib.rs:8:14
  |
8 |     let _ = c.clone();
  |              ^^^^^^^^ help: remove this
  |
  = note: `#[warn(clippy::redundant_clone)]` on by default
note: this value is dropped without further use
 --> src/lib.rs:8:13
  |
8 |     let _ = c.clone();
  |             ^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_clone
```

_Clippy outputs no warnings despite evidently having a redundant clone:_
```
    Finished dev [unoptimized + debuginfo] target(s) in 0.00s
```

# Meta
- `clippy 0.1.51 (04caa63 2021-01-30)`
- `rustc -Vv`:
```
rustc 1.51.0-nightly (04caa632d 2021-01-30)
binary: rustc
commit-hash: 04caa632dd10c2bf64b69524c7f9c4c30a436877
commit-date: 2021-01-30
host: x86_64-unknown-linux-gnu
release: 1.51.0-nightly
LLVM version: 11.0.1
```
